nixos/modules/services/x11/xserver.nix

```diff
@@ -652,15 +653,6 @@ in
                     || config.services.greetd.enable);
       in mkIf (default) (mkDefault true);
 
-    # so that the service won't be enabled when only startx is used
-    systemd.services.display-manager.enable  =
-      let dmConf = cfg.displayManager;
-          noDmUsed = !(dmConf.gdm.enable
-                    || dmConf.sddm.enable
-                    || dmConf.xpra.enable
-                    || dmConf.lightdm.enable);
-      in mkIf (noDmUsed) (mkDefault false);
-
     hardware.opengl.enable = mkDefault true;
```

Comment:
Should probably be moved to `displayManager` (or ideally an even more generic option). Same for XDG stuff, the inotify sysctls and the default target.

Reply:
See PR description